### PR TITLE
Remove RFC5246 updates, change Section 2 intro text such that the extension is always included.

### PR DESCRIPTION
**Pull request by chris-wood**
cc @jsalowey, @seanturner, @davidben

### CODE REVIEW

File: draft-ietf-tls-md5-sha1-deprecate-07.xml [hunk 1]
```diff
@@ -172,10 +172,10 @@
...
    <section anchor="Signature_algorithms" title="Signature Algorithms">
    <t>
-   Clients MUST NOT include MD5 and SHA-1 in the signature_algorithms extension. If
-   a client does not send a signature_algorithms extension, then the server
-   MUST abort the handshake and send a handshake_failure alert, except when
-   digital signatures are not used (for example, when using PSK ciphers).
+   Clients MUST include the signature_algorithms extension. Clients MUST NOT include MD5
+   and SHA-1 in it this extension. If a client does not send a signature_algorithms
+   extension, then the server MUST abort the handshake and send a handshake_failure alert,
+   except when digital signatures are not used (for example, when using PSK ciphers).
```
Comment (davidben) [inline draft-ietf-tls-md5-sha1-deprecate-07.xml]: "MUST abort ... except when .... are not used (for example ...)" is very hard to parse. :-)

There are also non-PSK examples like RSA cipher suites which... either do or don't look at signature_algorithms depending on how much you believe the server is supposed to check its own certificate chain. What if we moved this to the ServerKeyExchange bit and say:

> Servers MUST NOT include MD5 and SHA-1 in ServerKeyExchange messages. If no other signature algorithms are available (for example, if the client does not send a signature_algorithms extension), the server MUST abort the handshake with a handshake_failure alert or select a different cipher suite.

Comment (chris-wood) [reply draft-ietf-tls-md5-sha1-deprecate-07.xml]: Simpler -- I'll take it!

File: draft-ietf-tls-md5-sha1-deprecate-07.xml [hunk 2]
```diff
@@ -173,9 +173,10 @@
    <section anchor="Signature_algorithms" title="Signature Algorithms">
    <t>
    Clients MUST include the signature_algorithms extension. Clients MUST NOT include MD5
-   and SHA-1 in it this extension. If a client does not send a signature_algorithms
-   extension, then the server MUST abort the handshake and send a handshake_failure alert,
-   except when digital signatures are not used (for example, when using PSK ciphers).
+   and SHA-1 in it this extension. Servers MUST NOT include MD5 and SHA-1 in ServerKeyExchange 
+   messages. If no other signature algorithms are available (for example, if the client does 
+   not send a signature_algorithms extension), the server MUST abort the handshake with a 
+   handshake_failure alert or select a different cipher suite.
```
Comment (davidben) [inline draft-ietf-tls-md5-sha1-deprecate-07.xml]: Oh, I meant we could move it down to line 186 below, which has the side effect of implicitly conditioning on when you would send ServerKeyExchange. "Servers MUST NOT include..." is the first sentence of that section.

Comment (chris-wood) [reply draft-ietf-tls-md5-sha1-deprecate-07.xml]: Oh, duh. I forgot we had a section for that.

**Review by davidben — APPROVED** [review]

File: draft-ietf-tls-md5-sha1-deprecate-07.xml [hunk 3]
```diff
@@ -172,10 +172,8 @@
 
    <section anchor="Signature_algorithms" title="Signature Algorithms">
    <t>
-   Clients MUST NOT include MD5 and SHA-1 in the signature_algorithms extension. If
-   a client does not send a signature_algorithms extension, then the server
-   MUST abort the handshake and send a handshake_failure alert, except when
-   digital signatures are not used (for example, when using PSK ciphers).
+   Clients MUST include the signature_algorithms extension. Clients MUST NOT include MD5
+   and SHA-1 in it this extension.
```
Comment (davidben) [inline draft-ietf-tls-md5-sha1-deprecate-07.xml]: Oops, minor typo I noticed immediately afterwards:
```suggestion
   and SHA-1 in this extension.
```

**Review by ghedo — CHANGES_REQUESTED** [review]
I think you also need to update the abstract and introduction sections, as they both mention "This document updates ..." (same as https://github.com/tlswg/draft-ietf-tls-md5-sha1-deprecate/pull/18#pullrequestreview-746082313).  Will probably conflict with https://github.com/tlswg/draft-ietf-tls-md5-sha1-deprecate/pull/18 so we should merge that first.

**Comment by seanturner** [comment]
I believe this I-D should still include the "update: 5246 (if approved)" header, even though we took out the 5246 OLD/NEW section. We are still proposing that 5246 be changed to require signature_algorithms extension be sent and that MD5/SHA1 never be sent.

In other words, I think you can merge this PR assuming there are no conflicts.

**Review by seanturner — APPROVED** [review]
I believe this can be merged as is. I.e., don't remove the "udpates" header, change the abstract, or intro.
